test/ruby_lsp_rails/server_test.rb

```diff
@@ -176,10 +176,34 @@ def resolve_route_info(requirements)
     assert_equal("Hello\n", stderr)
   end
 
+  test "reloads if routes files change" do
+    old_routes = File.read("test/dummy/config/routes.rb")
+
+    @server.execute("route_location", { name: "user_path" })
+    location = response[:result][:location]
+    assert_match(%r{test/dummy/config/routes.rb:5$}, location)
+
+    File.write("test/dummy/config/routes.rb", <<~RUBY)
+      Rails.application.routes.draw do
+      end
+    RUBY
+
+    @server.execute("reload_routes", {})
+
+    @server.execute("route_location", { name: "user_path" })
+
+    assert_nil(response[:result])
+  ensure
+    File.write("test/dummy/config/routes.rb", old_routes)
+  end
+
   private
 
   def response
-    _headers, content = @stdout.string.split("\r\n\r\n")
-    JSON.parse(content, symbolize_names: true)
+    headers, content = @stdout.string.split("\r\n\r\n")
+    content_length = headers[/Content-Length: (\d+)/i, 1].to_i
+    # binding.break
```

Comment:
@vinistock I'm trying to figure out what's happening here: If I add the breakpoint, and call `@stdout.string = ""` each time, it passes. But I call it from the code it fails with `write': not opened for writing (IOError)`.